Mathlib/Order/SuccPred/Limit.lean

```diff
@@ -314,131 +347,189 @@ section Preorder
 
 variable [Preorder α] {a : α}
 
+/-- A predecessor limit is a value that isn't maximal and doesn't cover any other.
+
+It's so named because in a predecessor order, a predecessor limit can't be the predecessor of
+anything larger.
+
+This previously allowed the element to be maximal. This usage is now covered by `IsPredPreLimit`. -/
+def IsPredLimit (a : α) : Prop :=
+  ¬ IsMax a ∧ IsPredPrelimit a
+
+protected theorem IsPredLimit.not_isMax (h : IsPredLimit a) : ¬ IsMax a := h.1
+protected theorem IsPredLimit.isPredPrelimit (h : IsPredLimit a) : IsPredPrelimit a := h.2
+
+theorem IsPredLimit_iff' : IsPredLimit a ↔ ¬ IsMax a ∧ IsPredPrelimit a :=
+  Iff.rfl
+
+@[simp]
+theorem isSuccLimit_toDual_iff : IsSuccLimit (toDual a) ↔ IsPredLimit a := by
+  simp [IsSuccLimit, IsPredLimit]
```

Comment:
Is this definitional?
```suggestion
theorem isSuccLimit_toDual_iff : IsSuccLimit (toDual a) ↔ IsPredLimit a := .rfl
```

Reply:
Nope, since the underlying `toDual_covBy_toDual_iff` isn't definitional either.